spec/components/s3_helper_spec.rb

```diff
@@ -40,17 +46,25 @@
       rules = hash["LifecycleConfiguration"]["Rule"]
 
       expect(rules.length).to eq(2)
-
+      expect(rules[1]["Expiration"]["Days"]).to eq("100")
       # fixes the bad filter
       expect(rules[0]["Filter"]["Prefix"]).to eq("projectdocs/")
     end.to_return(status: 200, body: "", headers: {})
 
-    SiteSetting.enable_s3_uploads = true
-    SiteSetting.s3_access_key_id = "abc"
-    SiteSetting.s3_secret_access_key = "def"
+    helper = S3Helper.new('bob', 'tomb')
+    helper.update_tombstone_lifecycle(100)
+  end
+
+  it "can skip policy update when s3_configure_tombstone_policy is disabled" do
+    SiteSetting.s3_configure_tombstone_policy = false
+
+    hash = Hash.from_xml(@lifecycle)
```

Comment:
I am pretty sure you want to remove this as well and the assertion at the end. Try this without the site setting, it would explode in fire.